src/js/components/DateInput/DateInput.js

```diff
@@ -275,14 +278,20 @@ Use the icon prop instead.`,
                   reference,
                   outputFormat,
                 );
-                if (nextValue !== undefined)
-                  setReference(getReference(nextValue));
+
+                const validatedNextValue = validateDateBounds({
+                  dateBounds: bounds,
```

Comment:
```suggestion
                  dateBounds: calendarProps?.bounds,
```
Then we can remove the variable at line 169

Reply:
fixed 👍🏼